src/bun.js/bindings/CPUFeatures.zig

```diff
@@ -0,0 +1,81 @@
+const bun = @import("root").bun;
+const std = @import("std");
+
+fn Impl(comptime T: type) type {
+    return struct {
+        pub fn format(self: T, comptime _: []const u8, _: anytype, writer: anytype) !void {
+            var is_first = true;
+            inline for (comptime std.meta.fieldNames(T)) |fieldName| {
+                if (comptime bun.strings.eqlComptime(fieldName, "padding") or bun.strings.eqlComptime(fieldName, "none"))
+                    continue;
+
+                const value = @field(self, fieldName);
+                if (value) {
+                    if (!is_first)
+                        try writer.writeAll(" ");
+                    is_first = false;
+                    try writer.writeAll(fieldName);
+                }
+            }
+        }
+
+        pub fn isEmpty(self: T) bool {
+            return @as(u8, @bitCast(self)) == 0;
+        }
+
+        pub fn get() T {
+            return @bitCast(bun_cpu_features());
+        }
+    };
+}
+
+const X86CPUFeatures = packed struct(u8) {
+    none: bool = false,
```

Comment:
none should not be a flag here. the numeric values in c++ are off by one (you can start at zero, `1 << 0` is 1). then the padding bit increases by one

Reply:
this was deliberate, as a kind of sanity check

Reply:
add `assert(self.none == false and self.padding == 0);` to document this sanity check